### PR TITLE
[labs/virtualizer] handle uninitialized layout in _childrenSizeChanged

### LINKED ISSUE #3726: [labs/virtualizer]: uninitialized layout Error in _childrenSizeChanged
### Which package(s) are affected?

Virtualizer (@lit-labs/virtualizer)

### Description

 `Error` in `_childrenSizeChanged` when children is resized  before layout is initialized.

### Reproduction

Test case in PR.

### Workaround

None.

### Is this a regression?

No or unsure. This never worked, or I haven't tried before.

### Affected versions

1.0.1

### Browser/OS/Node environment

Any

### PR DESCRIPTION
**Pull request by megheaiulian**
Handle uninitialized layout in `_childrenSizeChanged`.

 Because the ResizeObserver starts listening very early it is possible that the layout has not been imported and initialized when a children has updated.

Fixes #3726

### CODE REVIEW

**Comment by usergenic** [comment]
Seems totally plausible.  Thanks for the fix, @megheaiulian

**Comment by justinfagnani** [comment]
@megheaiulian it would be great to have a test to prevent a regression here.

**Comment by usergenic** [comment]
The scenario where this would create a regression would be that the logic is relying on a weird unhandled runtime error to abort execution of the main childrenSizeChanged callback.  With or without this fix, the inner condition doesn't execute, but the "new" behavior is that code continues to execute without runtime error and invokes this._scheduleLayoutComplete(); etc--

I'll investigate the observer instantiation in relation to a deferred/dynamic layout assignment and see if there's a reason why this would behavior would be unwanted.

**Comment by usergenic** [comment]
i.e. it is an open question is whether this either creates a regression or fixes an edge case.

**Comment by graynorton** [comment]
Thanks for the PR, @megheaiulian!

A simplified repro case would definitely be helpful; at minimum, could you file an issue to match up with this PR?

@usergenic @justinfagnani There are unreleased changes in virtualizer that seem quite likely to be relevant, as they definitely touch (and aim to simplify) virtualizer initialization. Is there anything holding up a release? In any case, once we have a repro for this, we should make sure that it still holds when run against the code in `main`.

**Comment by megheaiulian** [comment]
@graynorton I will add a test for this.

**Comment by megheaiulian** [comment]
Add a test case to reproduce the error.

**Comment by megheaiulian** [comment]
@usergenic Could you please have a another look at this ? It is a very annoying little bug.

File: packages/labs/virtualizer/src/test/scenarios/immediate-sizing.test.ts
Comment (justinfagnani) [inline packages/labs/virtualizer/src/test/scenarios/immediate-sizing.test.ts:50]: This is causing a lint warning. Just don't declare a variable.

**Review by justinfagnani — CHANGES_REQUESTED** [review]
Looks good after the lint fix!

**Review by usergenic — APPROVED** [review]
Thank you for your patience on this and for adding the test case etc.  LGTM

**Comment by usergenic** [comment]
@justinfagnani this PR needs your approval because of prior requested change.

**Review by justinfagnani — APPROVED** [review]

**Comment by usergenic** [comment]
@megheaiulian alright this is finally merged.  Thanks again!

**Pull request merged**
